packages/extension/src/inpage/starknetWindowObject.ts

```diff
@@ -65,17 +72,34 @@ export const starknetWindowObject: StarknetWindowObject = {
     return waitForMessage("IS_PREAUTHORIZED_RES", 1000)
   },
   on: (event, handleEvent) => {
-    if (event !== "accountsChanged") {
+    if (event !== "accountsChanged" && event !== "networkChanged") {
       throw new Error(`Unknwown event: ${event}`)
```

Comment:
to avoid forgetting to deal with new events in future

```suggestion
      assertNever(event)
      throw new Error(`Unknwown event: ${event}`)
```

Reply:
Why do we need this?